### PR TITLE
Task/m schema for image nodes

### PR DESCRIPTION
**Pull request by abhimutant**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -226,7 +224,7 @@ func (c *Client) NewUnAuthRequest(ctx context.Context, method, urlStr string, bo
 			return nil, err
 		}
 	}
-
+	fmt.Println(buf.String())
```
Comment (bhati-pradeep) [inline client/client.go]: Can we remove this log/print ?

File: nutanix/helpers.go [hunk 2]
```diff
@@ -214,6 +216,24 @@ func taskStateRefreshFunc(client *v3.Client, taskUUID string) resource.StateRefr
 	}
 }
 
+func foundationImageRefresh(ctx context.Context, client *foundation.Client, taskUUID string) resource.StateRefreshFunc {
```
Comment (bhati-pradeep) [inline nutanix/helpers.go]: Can we change taskUUID to sessionID ?

File: nutanix/provider.go [hunk 3]
```diff
@@ -152,6 +152,7 @@ func Provider() *schema.Provider {
 			"nutanix_recovery_plan":           resourceNutanixRecoveryPlan(),
 			"nutanix_service_group":           resourceNutanixServiceGroup(),
 			"nutanix_address_group":           resourceNutanixAddressGroup(),
+			"nutanix_foundation_image_nodes":  ResourceFoundationImageNodes(),
```
Comment (bhati-pradeep) [inline nutanix/provider.go]:
```suggestion
			"nutanix_foundation_image_nodes":  resourceFoundationImageNodes(),
```
@siddharth-nutanix we keep every resource func private right ?

File: nutanix/resource_foundation_image_nodes.go [hunk 4]
```diff
@@ -0,0 +1,1173 @@
...
+	"github.com/spf13/cast"
+	"github.com/terraform-providers/terraform-provider-nutanix/client/foundation"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+var (
+	ImageMinTimeout = 60 * time.Minute
+)
+
+func ResourceFoundationImageNodes() *schema.Resource {
```
Comment (bhati-pradeep) [inline nutanix/resource_foundation_image_nodes.go]:
```suggestion
func resourceFoundationImageNodes() *schema.Resource {
```

**Review by bhati-pradeep — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (abhimutant) [reply client/client.go]: Done.

File: nutanix/helpers.go (hunk 2 above)
Comment (abhimutant) [reply nutanix/helpers.go]: changed.

File: nutanix/provider.go (hunk 3 above)
Comment (abhimutant) [reply nutanix/provider.go]: yes. I missed it .

**Review by bhati-pradeep — APPROVED** [review]

File: nutanix/resource_foundation_image_nodes.go [hunk 5]
```diff
@@ -0,0 +1,1171 @@
...
+			},
+			"session_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceFoundationImageNodesCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (siddharth-nutanix) [inline nutanix/resource_foundation_image_nodes.go:613]: Please use comma ok idiom for all type conversions.

File: nutanix/resource_foundation_image_nodes.go [hunk 6]
```diff
@@ -0,0 +1,1171 @@
...
+		Timeout: d.Timeout(schema.TimeoutCreate),
+		Delay:   1 * time.Minute,
+	}
+	info, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for image (%s) to be ready: %v", resp.SessionID, err)
+	}
+	if progress, ok := info.(*foundation.ImageNodesProgressResponse); ok {
+		if utils.Float64Value(progress.AggregatePercentComplete) < float64(AggregatePercentComplete) {
+			return diag.Errorf("Progress is incomplete")
```
Comment (siddharth-nutanix) [inline nutanix/resource_foundation_image_nodes.go:800]: Error should be more descriptive

**Review by siddharth-nutanix — APPROVED** [review]
Approved as we are merging to a feat branch. Resolve the comments in further PRs.

**Pull request merged**